.github/workflows/publish_snapshot.yml

```diff
@@ -0,0 +1,66 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+name: Publish Snapshot
+
+on:
+  schedule:
+    # At the end of every day
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+jobs:
+  publish-snapshot:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: snapshot-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            snapshot-maven-
+      - name: Publish snapshot
+        env:
+          ASF_USERNAME: ${{ secrets.NEXUS_USER }}
+          ASF_PASSWORD: ${{ secrets.NEXUS_PW }}
+        run: |
+          tmp_settings="tmp-settings.xml"
+          echo "<settings><servers><server>" > $tmp_settings
+          echo "<id>apache.snapshots.https</id><username>$ASF_USERNAME</username>" >> $tmp_settings
+          echo "<password>$ASF_PASSWORD</password>" >> $tmp_settings
+          echo "</server></servers></settings>" >> $tmp_settings
+          
+          set -o pipefail; mvn clean deploy -Dgpg.skip -DskipTests -Papache-release | tee ./mvn.log; set +o pipefail
+          if [[ $(grep -c "overlapping classes" ./mvn.log) -gt 0 ]];then
+            echo "Found overlapping classes, please fix"
+            exit 1
+          fi
+          if [[ $(git diff HEAD | wc -l) -gt 0 ]];then
+            echo "Please generate the java doc via 'mvn clean install -DskipTests -Pgenerate-docs' again"
+            exit 1
+          fi
```

Comment:
I think it would be better not to have these checks duplicated in all CI runs (we also might change some with the 1.15 bump)

Reply:
Fair, I will refactor it to a common function.

Reply:
That is also an option, I was suggesting to simply remove it. If it runs once somewhere else that is enough to generally fail the CI.

Reply:
Ok, going for the cheap option then.